wallet/src/Dscp/Wallet/KeyStorage.hs

```diff
@@ -0,0 +1,66 @@
+module Dscp.Wallet.KeyStorage
+       ( getAccounts
+       , addAccount
+       ) where
+
+import Data.Aeson (eitherDecode, encode)
+import Data.Aeson.TH (defaultOptions, deriveJSON)
+import Dscp.Core (mkAddr)
+import System.Directory (doesFileExist)
+import System.FileLock (SharedExclusive (..), withFileLock)
+
+import qualified Data.ByteString.Lazy as LBS
+
+import Dscp.Wallet.Face
+
+data Storage = Storage
+    { accounts :: [StorageAccount]
+    }
+
+data StorageAccount = StorageAccount
+    { secretKey :: Encrypted SecretKey
+    , publicKey :: PublicKey
+    }
+
+deriveJSON defaultOptions ''Storage
+deriveJSON defaultOptions ''StorageAccount
+
+storageFileName :: FilePath
+storageFileName = "wallet.json"
+
+storageLockFileName :: FilePath
+storageLockFileName = "wallet.json.lock"
+
+getAccounts :: IO [Account]
+getAccounts = do
+    storage <- readStorage
+    return $ toAccount <$> accounts storage
+  where
+    toAccount StorageAccount{..} = Account
+        { accountSecretKey = secretKey
+        , accountPublicKey = publicKey
+        , accountAddress = mkAddr publicKey
+        }
+
+addAccount :: Account -> IO ()
+addAccount account = modifyStorage (\(Storage accs) -> Storage $ accs ++ [fromAccount account])
+  where
+    fromAccount Account{..} = StorageAccount
+        { secretKey = accountSecretKey
+        , publicKey = accountPublicKey
+        }
+
+readOrCreateStorage :: IO Storage
+readOrCreateStorage = do
+    exists <- doesFileExist storageFileName
+    if exists
+        then LBS.readFile storageFileName >>= either fail return . eitherDecode
+        else return $ Storage []
+
+readStorage :: IO Storage
+readStorage = withFileLock storageLockFileName Shared $ \_ -> readOrCreateStorage
```

Comment:
Add `Versioned` wrapper over what you store on disk, so that we could change format later if needed.